src/cmd/promremotebench/main.go

```diff
@@ -97,30 +108,28 @@ func main() {
 		log.Println("simulating host", host.Name)
 	}
 
-	generateLoop(hostGen, *scrapeIntervalSeconds, 
+	generateLoop(hostGen, *scrapeIntervalSeconds, *newSeriesPercent,
 		client, *remoteBatchSize)
 }
 
 func generateLoop(
-	generator *generators.HostsSimulator, 
+	generator *generators.HostsSimulator,
 	intervalSeconds int,
-	remotePromClient *Client, 
+	newSeriesPercent float64,
+	remotePromClient *Client,
 	remotePromBatchSize int,
 ) {
-	period := time.Duration(intervalSeconds) * time.Second
-
-	ticker := time.NewTicker(period)
-	defer ticker.Stop()
+	series := generator.Generate(0, newSeriesPercent)
+	remoteWrite(series, remotePromClient, remotePromBatchSize)
 
-	// First with zero progression
-	remoteWrite(generator.Generate(0), remotePromClient,  
-	remotePromBatchSize)	
+	secTick := 1
+	for _ = range time.Tick(time.Second) {
+		go func(tick int) {
+			curTick := tick % intervalSeconds
+			series := generator.Generate(curTick, newSeriesPercent)
+			remoteWrite(series, remotePromClient, remotePromBatchSize)
+		}(secTick)
```

Comment:
Hm we should probably bound the number of active goroutines unless it just continually piles up and dies.

Perhaps just create a semi-on the fly worker pool here and simulate `GoIfAvailable`?

i.e.

```go
numWorkers := 1024
workers := make(chan struct{}, numWorkers)
for i := 0; i < numWorkers; i++ {
  workers <- struct{}{}
}

for ... { 
  select {
  case token := <-workers:
    go func () {
      defer workers <- token
      curTick := tick % intervalSeconds
      series := generator.Generate(curTick, newSeriesPercent)
      remoteWrite(series, remotePromClient, remotePromBatchSize)
    }()
  default:
    // too many active workers
  }
  tick++
}
```